packages/sdk/src/next/tests/controllers/VariableController.test.ts

```diff
@@ -0,0 +1,92 @@
+    };
+
+    beforeEach(() => {
+        mockedVariableController = new VariableController(mockEditorApi);
+        jest.spyOn(mockEditorApi, 'getVariableById');
+        jest.spyOn(mockEditorApi, 'getVariableByName');
+        jest.spyOn(mockEditorApi, 'getVariables');
+        jest.spyOn(mockEditorApi, 'setListVariableItems');
+    });
+
+    it('get variable by id', async () => {
+        const result = await mockedVariableController.getById('2');
+        expect(mockEditorApi.getVariableById).toHaveBeenCalledTimes(1);
+        expect(mockEditorApi.getVariableById).toHaveBeenCalledWith('2');
+        expect((result.parsedData as ListVariable).items).toStrictEqual([{ value: 'abc', displayValue: 'A-B-C' }]);
+    });
+
+    it('get variable by name', async () => {
+        const result = await mockedVariableController.getByName('name');
+        expect(mockEditorApi.getVariableByName).toHaveBeenCalledTimes(1);
+        expect(mockEditorApi.getVariableByName).toHaveBeenCalledWith('name');
+        // Backwards compatibility layer maps the new `VariableListItem` into a string.
+        expect((result.parsedData as ListVariable).items).toStrictEqual([{ value: 'abc', displayValue: 'A-B-C' }]);
+        expect((result.parsedData as ListVariable).selected).toStrictEqual({ value: 'abc', displayValue: 'A-B-C' });
+    });
+
+    it('get variable list', async () => {
+        const result = await mockedVariableController.getAll();
+        expect(mockEditorApi.getVariables).toHaveBeenCalledTimes(1);
+        // Backwards compatibility layer maps the new `VariableListItem` into a string.
+        expect((result.parsedData as ListVariable[])[0].items).toStrictEqual([{ value: 'abc', displayValue: 'A-B-C' }]);
+        expect((result.parsedData as ListVariable[])[0].selected).toStrictEqual({
+            value: 'abc',
+            displayValue: 'A-B-C',
+        });
+    });
+
+    it('sets the variable list items', async () => {
+        await mockedVariableController.setListVariable('listId', [{ value: 'a' }, { value: 'b' }, { value: 'c' }]);
+        expect(mockEditorApi.setListVariableItems).toHaveBeenCalledTimes(1);
+        expect(mockEditorApi.setListVariableItems).toHaveBeenCalledWith(
+            'listId',
+            (<ListVariableItem[]>[{ value: 'a' }, { value: 'b' }, { value: 'c' }]).map((item) => JSON.stringify(item)),
+        );
+
+        await mockedVariableController.setListVariable('listId', [
+            { value: 'a', displayValue: 'dis a' },
+            { value: 'b', displayValue: 'dis b' },
+            { value: 'c', displayValue: 'dis b' },
+        ]);
+        expect(mockEditorApi.setListVariableItems).toHaveBeenCalledTimes(2);
+        expect(mockEditorApi.setListVariableItems).toHaveBeenCalledWith(
+            'listId',
+            (<ListVariableItem[]>[{ value: 'a' }, { value: 'b' }, { value: 'c' }]).map((item) => JSON.stringify(item)),
```

Comment:
```suggestion
            (<ListVariableItem[]>[{ value: 'a', displayValue: 'dis a' }, { value: 'b', displayValue: 'dis b' }, { value: 'c', displayValue: 'dis c' }]).map((item) => JSON.stringify(item)),
```

Reply:
how does he not even fail on this

Reply:
Because the first `setListVariable` call produces this result